src/lib/core/ErrorStr.cpp

```diff
@@ -50,8 +50,25 @@ static ErrorFormatter * sErrorFormatterList = nullptr;
  */
 DLL_EXPORT const char * ErrorStr(CHIP_ERROR err, bool withSourceLocation)
 {
-    char * formattedError   = sErrorStr;
-    uint16_t formattedSpace = sizeof(sErrorStr);
+    ErrorStr(err, withSourceLocation, sErrorStr, sizeof(sErrorStr));
+    return sErrorStr;
+}
+
+/**
+ * This routine writess a human-readable NULL-terminated C string into the buf
+ * which describes the provided error.
+ *
+ * @param[in] err                      The error for format and describe.
+ * @param[in] withSourceLocation       Whether or not to include the source
+ * @param[in] buf                      Provided buf to write into
+ * @param[in] bufSize                  Size of the buf
+ * location in the output string. Only used if CHIP_CONFIG_ERROR_SOURCE &&
+ * !CHIP_CONFIG_SHORT_ERROR_STR. Defaults to true.
+ */
+DLL_EXPORT void ErrorStr(CHIP_ERROR err, bool withSourceLocation, char * buf, uint16_t bufSize)
+{
+    char * formattedError   = buf;
+    uint16_t formattedSpace = sizeof(buf);
```

Comment:
this seems to be sizeof(char *) ... is this right?

Reply:
Oops, you are right.